src/vdbfusion_ros/VDBVolume_ros.cpp

```diff
@@ -107,19 +127,41 @@ bool vdbfusion::VDBVolumeNode::saveVDBVolume(vdbfusion_ros::save_vdb_volume::Req
     openvdb::io::File(volume_name + "_grid.vdb").write({vdb_volume_.tsdf_});
 
     // Run marching cubes and save a .ply file
-    auto [vertices, triangles] =
+    auto [vertices, triangles, colors] =
         this->vdb_volume_.ExtractTriangleMesh(this->fill_holes_, this->min_weight_);
 
-    Eigen::MatrixXd V(vertices.size(), 3);
+
+    // Suppose these hold your data
+    std::vector<std::array<double, 3>> meshVertexPositions;
     for (size_t i = 0; i < vertices.size(); i++) {
-        V.row(i) = Eigen::VectorXd::Map(&vertices[i][0], vertices[i].size());
+        std::array<double, 3> vertex = {vertices[i][0], vertices[i][1], vertices[i][2]};
+        meshVertexPositions.emplace_back(vertex);
     }
 
-    Eigen::MatrixXi F(triangles.size(), 3);
+    std::vector<std::vector<size_t>> meshFaceIndices;
     for (size_t i = 0; i < triangles.size(); i++) {
-        F.row(i) = Eigen::VectorXi::Map(&triangles[i][0], triangles[i].size());
+        std::vector<size_t> triangle = {triangles[i][0], triangles[i][1], triangles[i][2]};
```

Comment:
Replace with:
`std::vector<size_t> triangle = {static_cast<size_t>(triangles[i][0]), static_cast<size_t>(triangles[i][1]), static_cast<size_t>(triangles[i][2])};`

Throws compile-time error as the types don't match
`triangles[i][j]` returns an integer whereas `size_t` is of type `unsigned int`